peer/agent.go

```diff
@@ -18,24 +18,22 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package transport
+package peer
 
-//go:generate mockgen -destination=transporttest/peeragent.go -package=transporttest go.uber.org/yarpc/transport Agent,PeerSubscriber
+//go:generate mockgen -destination=peertest/agent.go -package=peertest go.uber.org/yarpc/peer Agent,Subscriber
 
-// PeerSubscriber listens to changes of a Peer over time.
-type PeerSubscriber interface {
-	// The Peer Notifies the PeerSubscriber when its status changes (e.g. connections status, pending requests)
-	NotifyStatusChanged(Peer)
+// Subscriber listens to changes of a Peer over time.
+type Subscriber interface {
+	// The Peer Notifies the Subscriber when its status changes (e.g. connections status, pending requests)
+	NotifyStatusChanged(Identifier)
 }
 
-// Agent manages Peers across different PeerSubscribers.  A PeerSubscriber will request a Peer for a specific
+// Agent manages Peers across different Subscribers.  A Subscriber will request a Peer for a specific
 // PeerIdentifier and the Agent has the ability to create a new Peer or return an existing one.
 type Agent interface {
-	PeerSubscriber
+	// Get or create a Peer for the Subscriber
+	RetainPeer(Identifier, Subscriber) (Peer, error)
```

Comment:
The naming on this seems funny... why not just call it `GetPeer` where the semantics are get or create?